### ISSUE TITLE
Splat not working at all when wilcard is an optional parameter

### ISSUE THREAD

**Issue by kouts** [mikecao/flight#144]
Imagine a route like this:

```
Flight::route('/@name/account(/*)', function($name){
    echo "hello";
});
```

or even this

```
Flight::route('/account(/*)', function(){
    echo "hello";
});
```

Splat is empty because it looks at the last character which may as well be a parenthesis...

**Comment by mikecao** [comment]
The splat already accounts for additional url segments so you don't need to declare it as optional. It will match both `/account/` and `/account/and/more`. Optional parameters can only be used with named parameters.

**Comment by kouts** [comment]
It would be useful if you don't want to declare for example:

```
Flight::route('/account', function(){
    echo "hello";
});
Flight::route('/account/*', function(){
    echo "hello";
});
```

twice.
Please think about it.
At the moment optional wildcard works, it's just that the splat isn't calculated correctly.

**Comment by mikecao** [comment]
It seems like your issue is with the trailing slash. I think I'll make an adjustment to the regex to account for that when using a splat.

**Comment by mikecao** [comment]
Fixed in commit 64d384ddffc5bc23ca641d313214974b8b0a5788.

**Comment by kouts** [comment]
Ok thanks, it now works without optional wildcard.
